### PR TITLE
[MM-42366] Add summary to the playbook editor

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR adds summary to the playbook editor

https://user-images.githubusercontent.com/2340127/165731331-11e8b558-7831-460e-acb9-eb98182f2d43.mp4

#### Ticket Link

  Fixes: https://mattermost.atlassian.net/browse/MM-42366

#### Checklist

- ~~[ ] Gated by experimental feature flag~~
- ~~[ ] Unit tests updated~~

### CODE REVIEW

**Comment by iomodo** [comment]
Merged with master, new UI:

https://user-images.githubusercontent.com/2340127/166262218-df8889ca-b867-46d6-8946-91f5b8afe19a.mp4

File: webapp/src/components/backstage/playbook_editor/outline/outline.tsx [hunk 1]
```diff
@@ -27,14 +29,35 @@ interface Props {
 type Attrs = HTMLAttributes<HTMLElement>;
 
 /** @alpha replace/copy-pasta/unfold sections as-needed*/
-const Outline = ({playbook}: Props) => {
+const Outline = (props: Props) => {
     const {formatMessage} = useIntl();
+    const [playbook, setPlaybook] = useState(props.playbook);
+
+    const updateSummaryForPlaybook = (summary: string) => {
```
Comment (crspeller) [inline webapp/src/components/backstage/playbook_editor/outline/outline.tsx:36]: I will fix this up with my PR.

**Review by crspeller — APPROVED** [review]
Looks nice!

**Review by calebroseland — APPROVED** [review]

**Comment by iomodo** [comment]
Merging this as status updates PR is depending on this. @abhijit-singh, we can fix UI in the next PR, since this PR is not changing the current flow

**Pull request merged**